src/validation.cpp

```diff
@@ -1669,6 +1748,41 @@ class WarningBitsConditionChecker : public AbstractThresholdConditionChecker
 // Protected by cs_main
 static ThresholdConditionCache warningcache[VERSIONBITS_NUM_BITS];
 
+static unsigned int GetBlockScriptFlags(const CBlockIndex* pindex, const CChainParams& chainparams) {
```

Comment:
I noted that at the beginning "I know not all flags are activated with bip9", but I still think it's the best place for this new function (even if some lines of the function don't rely on versionbits, they also don't rely on anything that versionbits didn't already rely on).Doing a separate module only for this single function seems stupid to me.
Anyway, as disappointing as your rejection to move the new function for free it is to me, let's move on. 

Can you at least avoid the use of globals you don't need to use, ie versionbitscache and cs_main  (both used directly and indirectly through IsWitnessEnabled())?
That way, if we ever move this function out of validation (like all previous attempts at creating it did) it can be a clean moveonly commit (whether it goes to versionbits.o or somewhere else) instead of needing to change the function signature.